src/main/java/com/fcgl/madrid/points/service/UserActionService.java

```diff
@@ -1,22 +1,70 @@
 package com.fcgl.madrid.points.service;
 
+import com.fcgl.madrid.points.model.Action;
+import com.fcgl.madrid.points.model.Trophy;
 import com.fcgl.madrid.points.model.UserAction;
+import com.fcgl.madrid.points.model.UserTrophy;
+import com.fcgl.madrid.points.repository.ActionRepository;
+import com.fcgl.madrid.points.repository.TrophyRepository;
 import com.fcgl.madrid.points.repository.UserActionRepository;
+import com.fcgl.madrid.points.repository.UserTrophyRepository;
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.stereotype.Service;
 
 import java.util.List;
 
 @Service
 public class UserActionService {
     private UserActionRepository userActionRepository;
+    private ActionRepository actionRepository; // Is this the correct way to access the action table?
+    private TrophyRepository trophyRepository;
+    private UserTrophyRepository userTrophyRepository;
 
     @Autowired
-    public UserActionService(UserActionRepository userActionRepository) {
+    public UserActionService(UserActionRepository userActionRepository,
+                             ActionRepository actionRepository,
+                             TrophyRepository trophyRepository,
+                             UserTrophyRepository userTrophyRepository) {
         this.userActionRepository = userActionRepository;
+        this.actionRepository = actionRepository;
+        this.trophyRepository = trophyRepository;
+        this.userTrophyRepository = userTrophyRepository;
     }
 
     public List<UserAction> findAll() {
         return userActionRepository.findAll();
     }
+
+    public ResponseEntity<Trophy> addUserAction(Long userID, Long actionID) {
+
+        // Check that the action exists
+        Action action = actionRepository.findById(actionID).orElse(null);
+        if (action == null) {
+            return ResponseEntity.badRequest().body(null);
```

Comment:
You asked what `InternalStatus` adds. And we wanted to have a simple places to store return messages that we could manage ourselves. It will be useful to have for when we add the translation system here

Reply:
@AshleeCW adds more functionality to `InternalStatus` in his pending PR which I am going to test after reviewing this PR

Reply:
Ok good!